### PR TITLE
Add Icon Component

### LINKED ISSUE #2170: Create IconComponent
**Describe the problem/challenge you have**
At the moment there is no way to create an Icon, It will be nice to have be able to use the clarity icons https://clarity.design/foundation/icons/api/

**Describe the solution you'd like**
Create IconComponent

**Anything else you would like to add:**
Necessary for issue #2018

**Environment:**

- Octant version (use `octant version`): 
- Kubernetes version (use `kubectl version`):
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc):

### PR DESCRIPTION
**Pull request by lenriquez**
Signed-off-by: Luis Enriquez <felipe096@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**
- Fixes #2170 

**Special notes for your reviewer**:
We are using this API 
https://v4.clarity.design/icons/how-to-use
and not this 
https://clarity.design/foundation/icons/api/
**Release note**:
```
release-note
```

### CODE REVIEW

**Comment by lenriquez** [comment]
This should be ready for review

**Review by wwitzel3 — CHANGES_REQUESTED** [review]
This isn't quite right. The badge isn't rendering properly.

```
{
  "metadata": {
    "type": "icon"
  },
  "config": {
    "shape": "bell",
    "size": "22",
    "solid": true,
    "status": "none",
    "inverse": false,
    "badge": "danger"
  }
}
```

Gives me: 
image

I'd expect, a red danger badge but in this case, whole icon is red.
image

Also I think `status` is too specific. Lets go with color and we can make some static types for our Success, Warning, and Failure status colors.

**Comment by wwitzel3** [comment]
Another question, is this component useful as a stand-alone component like this? Or do we want to extend the other components to support an optional Icon as well?

For example if I want to write something like `My Link <Icon>` .. not sure how we do that with this currently?

I know this PR is just implementing the raw component, but I want to make sure we think about this as we build other things that use the icon component.

**Comment by lenriquez** [comment]
@wwitzel3 Thanks for finding the bug! I will fix this tomorrow
I was thinking not on extending but rather on built a component that put the two component together. Something like "decorateComponent" and add the link and Icon to it `decorateComponent(Link, Icon)` with some configuration to show where the icon should be (left or right) and maybe the space should use. I am not sure about the name of the new component yet

**Comment by GuessWhoSamFoo** [comment]
Should we hold this until we can upgrade to Clarity 5?

File: web/src/app/modules/shared/components/presentation/icon/icon.component.ts
Comment (wwitzel3) [inline web/src/app/modules/shared/components/presentation/icon/icon.component.ts:24]: After this merges, please create an issue that references all of these locations we can remove these mappings once we migrate from `clr-icon` to `cds-icon`

File: web/src/app/modules/shared/models/content.ts
Comment (wwitzel3) [inline web/src/app/modules/shared/models/content.ts:596]: I want to change this field to `color` and then create some static helpers called `SuccessColor` , `WarningColor` , `DangerColor` .. I think limiting this to just for colors will be a problem.

**Review by wwitzel3 — CHANGES_REQUESTED** [review]

File: web/src/app/modules/shared/components/presentation/icon/icon.component.ts
Comment (lenriquez) [reply web/src/app/modules/shared/components/presentation/icon/icon.component.ts:24]: Sure!

File: web/src/app/modules/shared/models/content.ts
Comment (lenriquez) [reply web/src/app/modules/shared/models/content.ts:596]: by static helper do you mean change the mapper for a switch case? I mean 
rather than this
```
 statusHash = {
    info: 'is-info ',
    success: 'is-success ',
    warning: 'is-warning ',
    danger: 'is-error ',
  };
   mapStatus(status: string) {
    if (!status || status === '') {
      return;
    }
    this.classAttr += this.statusHash[status] || '';
  }
```
use this

```

  static SuccessColor = 'is-success';

  mapColor() {
    switch () {
      case: 'success'
        return this.SuccessColor;
    }
    .
    .
     .
  }

```

Comment (lenriquez) [reply web/src/app/modules/shared/models/content.ts:596]: BTW Clarity 5 does a property call colors
https://clarity.design/foundation/icons/api/
image

Comment (lenriquez) [reply web/src/app/modules/shared/models/content.ts:596]: Done!
image

Comment (wwitzel3) [reply web/src/app/modules/shared/models/content.ts:596]: If both `color` and `status` are set, ignore `color` and use `status` .. other than that, this is great!

Comment (lenriquez) [reply web/src/app/modules/shared/models/content.ts:596]: ohh I misunderstood that, It should work now
image

**Review by wwitzel3 — APPROVED** [review]

**Pull request merged**
